Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/MenuViewController.swift

```diff
@@ -1,405 +1,287 @@
-        
-        tableView.showsVerticalScrollIndicator = false
-        tableView.isScrollEnabled = false
-        
-        // Hide separator line of the last cell.
-        tableView.tableFooterView =
-            UIView(frame: CGRect(x: 0, y: 0, width: tableView.frame.size.width, height: 1))
-        
-        let size = CGSize(width: 200, height: UIScreen.main.bounds.height)
-        
-        let fit = view.systemLayoutSizeFitting(
-            size,
-            withHorizontalFittingPriority: .required,
-            verticalFittingPriority: .defaultHigh
-        )
-        
-        preferredContentSize = CGSize(width: fit.width, height: fit.height + UX.topBottomInset * 2)
-        
-        NotificationCenter.default.addObserver(self, selector: #selector(vpnConfigChanged),
-                                               name: .NEVPNStatusDidChange, object: nil)
+    override var shouldAutorotate: Bool {
+        // Due to a bug in PanModal, the presenting controller does not receive safe area updates
+        // while a pan modal is presented, therefore for the time being, do not allow rotation
+        // while this menu is open.
+        //
+        // Issue: https://github.com/slackhq/PanModal/issues/139
+        false
     }
     
-    deinit {
-        NotificationCenter.default.removeObserver(self)
-    }
-    
-    // MARK: - Table view data source
+    private var isDismissing: Bool = false
     
-    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        tableView.deselectRow(at: indexPath, animated: false)
-        
-        guard let cell = tableView.cellForRow(at: indexPath) else { return }
-        
-        guard let button = MenuButtons(rawValue: cell.tag) else {
-            assertionFailure("No cell with \(cell.tag) tag.")
-            return
-        }
-        
-        switch button {
-        case .vpn:
-            guard let menuCell = cell as? MenuCell else { return }
-            openVPNAction(menuCell: menuCell)
-        case .bookmarks: openBookmarks()
-        case .history: openHistory()
-        case .settings: openSettings()
-        case .add: openAddBookmark()
-        case .share: openShareSheet()
-        case .downloads: openDownloads()
+    override func dismiss(animated flag: Bool, completion: (() -> Void)? = nil) {
+        if let _ = presentedViewController as? InnerMenuNavigationController,
+           presentingViewController?.presentedViewController === self {
+            isDismissing = true
+            presentingViewController?.dismiss(animated: flag, completion: completion)
+        } else {
+            super.dismiss(animated: flag, completion: completion)
         }
     }
     
-    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return visibleButtons.count
+    private var isPresentingInnerMenu: Bool {
+        if let _ = presentedViewController as? InnerMenuNavigationController {
```

Comment:
can we use is here instead 
`presentedViewController is InnerMenuNavigationController`